pulsar/internal/connection_pool.go

```diff
@@ -84,9 +87,11 @@ func NewConnectionPool(
 	return p
 }
 
-func (p *connectionPool) GetConnection(logicalAddr *url.URL, physicalAddr *url.URL) (Connection, error) {
-	p.log.WithField("logicalAddr", logicalAddr).WithField("physicalAddr", physicalAddr).Debug("Getting pooled connection")
-	key := p.getMapKey(logicalAddr, physicalAddr)
+func (p *connectionPool) GetConnection(logicalAddr *url.URL, physicalAddr *url.URL,
+	keySuffix int32) (Connection, error) {
+	p.log.WithField("logicalAddr", logicalAddr).
+		WithField("physicalAddr", physicalAddr).Debug("Getting pooled connection")
```

Comment:
```suggestion
	p.log.WithField("logicalAddr", logicalAddr).
		WithField("physicalAddr", physicalAddr).
		WithField("keySuffix", keySuffix).Debug("Getting pooled connection")
```
Better to add a new field `keySuffix` to help debug.